### DISCUSSION TITLE
How to set custom theme to external url with marp-cli

### DISCUSSION THREAD

**Discussion by SoumayaMauthoorMOJ** [marp-team/marp#523] [Q&A] [answered]
marp-vscode allows you to specify both relative paths and remote URLs as custom themes (see [here](https://github.com/marp-team/marp-vscode?tab=readme-ov-file#use-custom-theme-css-%EF%B8%8F)):

```
// Please put `.vscode/settings.json` on your workspace
{
  "markdown.marp.themes": [
    "https://example.com/foo/bar/custom-theme.css",
    "./themes/your-theme.css"
  ]
}
```

marp-cli allows you to specify a relative path to a custom theme (see [here](https://github.com/marp-team/marp-cli?tab=readme-ov-file#use-custom-theme):

```
marp --theme custom-theme.css
```

I've tried passing an external url e.g. through [jsdeliver](https://cdn.jsdelivr.net/gh/ministryofjustice/marp-moj-theme/themes/moj.css) but no luck. Do I need to download the css file first, for example as part of a GitHub action?

**Comment by yhatt** [comment, marked as answer]
Yes, you should download the theme from remote before.

```
wget https://example.com/foo/bar/custom-theme.css
marp --theme custom-theme.css
```

Marp for VS Code allows remote URL in `markdown.marp.themes` setting that have [compatibillity with custom style settings `markdown.styles` for VS Code Markdown preview](https://code.visualstudio.com/docs/languages/markdown#_using-your-own-css), but Marp CLI does not allow passing themes as external URL.

Please note that Marp team is not leading implementation about this function, _to make CLI secure by reducing remote access capabilities._ However, the pull request from the community has been reviewed and in progress.

- https://github.com/marp-team/marp-cli/issues/183
- https://github.com/marp-team/marp-cli/pull/571

**Reply by SoumayaMauthoorMOJ** [reply]
Thanks!
